src/Archives/AdaptiveHuffmanTree.cpp

```diff
@@ -1,36 +1,32 @@
 #include "AdaptiveHuffmanTree.h"
+#include <utility>
 #include <stdexcept>
 
 namespace Archives
 {
 	// Creates an (adaptive) Huffman tree with numTerminalNodes at the bottom.
-	AdaptiveHuffmanTree::AdaptiveHuffmanTree(unsigned short numTerminalNodes)
-	{
-		unsigned short i;
-		unsigned short left;
-
+	AdaptiveHuffmanTree::AdaptiveHuffmanTree(NodeType numTerminalNodes) :
```

Comment:
What do you think about updating this variable name to terminalNodeCount?

All the class level variables could be initialized using constructor initialization list. I'm not sure if there is a preference one way or the other here. From my limited point of view, I think it is better to use the the initialization list?

Reply:
I prefer not using initialization lists, except where required. They are order dependent in non-intuitive ways. For instance, you may notice how that commit re-ordered the declaration of member variables to match initialization order.

At any rate, all member variables are currently initialized in the constructor initializer list. Or at least, as far as they can be. The vectors have space allocated in the initializer list. Initializing the values of the vectors is too complicated to be done in an initializer, so they must be done in the body of the constructor.